lib/r10k/cli/deploy.rb

```diff
@@ -23,6 +23,8 @@ def self.command
 
         required :c, :config, 'Specify a configuration file'
 
+        required nil, :cachedir, 'Specify a cachedir, overriding the value in config'
```

Comment:
https://tickets.puppetlabs.com/browse/RK-120 covers making `--config` global, and all that really entails is moving the settings loading/initialization from the `Action::Deploy::*` classes to https://github.com/puppetlabs/r10k/blob/master/lib/r10k/action/runner.rb#L39-L41, and then passing the loaded configuration to the called action classes.

What we could do is merge this as-is so that we can continue work on the things that depend on this PR, and once this is in we can also work on RK-120. Does that work for you?